### PR TITLE
[OV JS] Remove extra assets from tests

### PR DESCRIPTION
**Pull request by almilosz**
### Details:
 - Remove testing models from repository
 - Add `setup.js` script that will be called once to download testing models
 - Before running the suite check if model is downloaded, add top describe to *.test.js files

### Notes:
Currently there is no global setup file for native [node test runner](https://github.com/nodejs/node/issues/49732) as in other test runner e.g. [jest](https://jestjs.io/docs/configuration#globalsetup-string)

### Tickets:
 - [CVS-146345](https://jira.devtools.intel.com/browse/CVS-146345)

### CODE REVIEW

File: src/bindings/js/node/scripts/download_runtime.js [hunk 1]
```diff
@@ -281,4 +281,4 @@ function unarchive(tarFilePath, dest) {
   });
 }
 
-module.exports = { downloadRuntime };
+module.exports = { downloadRuntime, downloadFile, checkIfDirectoryExists };
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/scripts/download_runtime.js:284]: You need to rename function `checkIfDirectoryExists ` if you check not directory existence

File: src/bindings/js/node/tests/unit/basic.test.js [hunk 2]
```diff
@@ -4,329 +4,336 @@
...
-
-describe('Core.getVersions()', () => {
-
-  it('getVersions(validDeviceName: string)', () => {
-    const deviceVersion = core.getVersions('CPU');
-    assert.strictEqual(typeof deviceVersion, 'object');
-    assert.strictEqual(typeof deviceVersion.CPU, 'object');
-    assert.strictEqual(typeof deviceVersion.CPU.buildNumber, 'string');
-    assert.strictEqual(typeof deviceVersion.CPU.description, 'string');
+  before( async () => {
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/unit/basic.test.js:12]:
```suggestion
  before(async () => {
```

File: src/bindings/js/node/tests/unit/utils.js [hunk 3]
```diff
@@ -3,15 +3,56 @@
...
+  if ( !modelExists ) {
+    const { env } = process;
+    const proxyUrl = env.http_proxy || env.HTTP_PROXY || env.npm_config_proxy;
+
+    await downloadFile(model.xmlURL, baseArtifactsDir, model.xml, proxyUrl);
+    await downloadFile(model.binURL, baseArtifactsDir, model.bin, proxyUrl);
+  }
+}
+
+async function checkTestModel(model) {
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/unit/utils.js:50]: What does this function check? How about names: `isModelPresent` or `isModelAvailable`?

File: src/bindings/js/node/tests/unit/pre_post_processor.test.js [hunk 4]
```diff
@@ -4,161 +4,167 @@
...
-const core = new ov.Core();
-const model = core.readModelSync(testXml);
+describe('ov.preprocess.PrePostProcessor tests', () => {
 
-describe('PrePostProcess', () => {
-
-  it('input() ', () => {
-    assert.doesNotThrow(() =>
-      new ov.preprocess.PrePostProcessor(model).input());
+  before( async () => {
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/unit/pre_post_processor.test.js:12]:
```suggestion
  before(async () => {
```

File: src/bindings/js/node/tests/unit/read_model.test.js [hunk 5]
```diff
@@ -5,81 +5,90 @@
...
-    assert.equal(model.inputs.length, 1);
-  });
+describe('Tests for reading model.', () => {
 
-  it('readModeSync throws', () => {
-    assert.throws(
-      () => core.readModelSync(core),
-      /'readModelSync' method called with incorrect parameters./,
-    )
+  before( async () => {
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/unit/read_model.test.js:15]:
```suggestion
  before(async () => {
```

File: src/bindings/js/node/tests/unit/setup.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+const { testModels, downloadTestModel } = require('./utils.js');
+
+async function main() {
+  await downloadTestModel(testModels.testModelFP32);
+}
+
+if (require.main === module) {
+  main();
+}
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/unit/setup.js:9]: You can move it above of main definition

File: src/bindings/js/node/tests/unit/utils.js [hunk 7]
```diff
@@ -3,15 +3,68 @@
...
+    const proxyUrl = env.http_proxy || env.HTTP_PROXY || env.npm_config_proxy;
+
+    await downloadFile(model.xmlURL, modelsDir, model.xml, proxyUrl);
+    await downloadFile(model.binURL, modelsDir, model.bin, proxyUrl);
+  } catch(error) {
+    console.error(`Failed to download the model: ${error}.`);
+    throw error;
+  }
+
+}
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/unit/utils.js:59]:
```suggestion
  }
}
```

File: src/bindings/js/node/tests/unit/utils.js [hunk 8]
```diff
@@ -3,15 +3,68 @@
...
+  const modelsDir = './tests/unit/test_models';
+  try {
+    const ifModelsDirectoryExists = await checkIfPathExists(modelsDir);
+    if (!ifModelsDirectoryExists) {
+      await fs.mkdir(modelDir);
+    }
+
+    const modelPath = path.join(modelsDir, model.xml);
+    const modelExists = await checkIfPathExists(modelPath);
+    if ( modelExists ) return;
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/unit/utils.js:47]:
```suggestion
    if (modelExists) return;
```

File: src/bindings/js/node/tests/unit/utils.js [hunk 9]
```diff
@@ -3,15 +3,68 @@
...
+    throw error;
+  }
+
+}
+
+async function isModelAvailable(model) {
+  const baseArtifactsDir = './tests/unit/test_models';
+  const modelPath = path.join(baseArtifactsDir, model.xml);
+  const modelExists = await checkIfPathExists(modelPath);
+  if ( modelExists ) return;
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/unit/utils.js:65]:
```suggestion
  if (modelExists) return;
```

File: src/bindings/js/node/tests/unit/infer_request.test.js [hunk 10]
```diff
@@ -4,277 +4,286 @@
...
-  tensorLike.forEach(([tl]) => {
-    const result = inferRequest.infer([tl]);
-    const label = tl instanceof Float32Array ? 'TypedArray[]' : 'Tensor[]';
-    it(`Test infer(inputData: ${label})`, () => {
-      assert.deepStrictEqual(Object.keys(result), ['fc_out']);
-      assert.deepStrictEqual(result['fc_out'].data.length, 10);
-    });
+describe('ov.InferRequest tests', () => {
 
+  before( async () => {
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/unit/infer_request.test.js:15]:
```suggestion
  before(async () => {
```

**Review by vishniakov-nikolai — APPROVED** [review]

File: src/bindings/js/node/tests/unit/compiled_model.test.js [hunk 11]
```diff
@@ -4,75 +4,82 @@
...
-        /'setProperty' method called with incorrect parameters/
-      );
+  const testXml = getModelPath().xml;
+  const core = new ov.Core();
+  const properties = {
+    'AUTO_BATCH_TIMEOUT': '1',
+  };
+  const compiledModel = core.compileModelSync(testXml, 'BATCH:CPU', properties);
+
+  describe('setProperty() / getProperty()', () => {
```
[the remark this answers is not in the document]
Comment (almilosz) [reply src/bindings/js/node/tests/unit/compiled_model.test.js:23]: Removed

**Review by vishniakov-nikolai — APPROVED** [review]
Brilliant! LGTM

**Pull request merged**